homeassistant/components/frigidaire/__init__.py

```diff
@@ -0,0 +1,34 @@
+"""The frigidaire integration."""
+from __future__ import annotations
+
+import frigidaire
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+
+from .const import DOMAIN, PLATFORMS
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up frigidaire from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    def setup(username: str, password: str) -> frigidaire.Frigidaire:
+        hass.data[DOMAIN][entry.entry_id] = frigidaire.Frigidaire(username, password)
```

Comment:
Looking at the upstream code, it seems like this method can throw exceptions.
We should catch those here and throw a ConfigEntryNotReady in case of connection errors and for authentication errors, we should either abort or trigger reauthentication.